src/main/java/org/candlepin/resteasy/JsonProvider.java

```diff
@@ -147,14 +147,16 @@ public JsonProvider(boolean indentJson) {
             null));
         // Ensure our DateSerializer is used for all Date objects
         customModule.addSerializer(Date.class, new DateSerializer());
+
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        configureHateoasObjectMapper(mapper, indentJson);
+
         // Ensure we handle releaseVer fields properly
         customModule.addDeserializer(ReleaseVerDTO.class, new ReleaseVersionWrapDeserializer());
         customModule.addDeserializer(ConsumerTypeDTO.class, new ConsumerTypeDeserializer());
-        customModule.addDeserializer(GuestIdDTO.class, new GuestIdDeserializer());
+        customModule.addDeserializer(GuestIdDTO.class, new GuestIdDeserializer(mapper.reader()));
```

Comment:
It appears that we cannot use `ObjectMapper.readerFor()` because the ObjectMapper has a TypeFactory that caches the class type and this is referenced in `ObjectMapper.registerModule(customModule)` and breaks the custom deserializers.